### DISCUSSION TITLE
Capire chi utilizza la porta 80

### DISCUSSION THREAD

**Discussion by Paoloc66** [indomus/forum#2484] [HUB: Home Assistant]
ciao a tutti, ho questo problema:
devo rinnovare il certificato Let's encrypt, come ogni 3 mesi apro la porta 80 sul router e lancio il componente. 
Questa volta mi dice che la porta 80 è già utilizzata, ma non riesco a trovare chi la sta utilizzando.
Esiste qualche comando o metodo per scoprirlo?
Grazie

**Comment by SanBitter79** [comment]
Ciao,
assumendo che si tratti di un ambiente Linux puoi usare netstat:
***netstat -anpe | grep "80" | grep "LISTEN"***
Non entro nel merito del volersi ricordare di eseguire ogni 3 mesi dei task manuali per poter lanciare un'automazione

**Comment by Paoloc66** [comment]
Grazie, anche se purtroppo non ha trovato nulla.

Riguardo al task manuale è dovuto al fatto che devo aprire la porta col redirect sul router. E non penso di riuscire a farlo automatizzato

**Comment by SanBitter79** [comment]
Immagino che avrai un proxy per ruotare il traffico mappato sul router e proveniente da pubblica verso l'istanza locale HA. Avrai anche un port mapping verso HTTP/HTTPS dello stesso proxy per gestire il rinnovo dei certificati di Let'sEncrypt. Se è così puoi lasciare le regole sempre attive sul firewall e bloccare sul proxy le connessioni HTTP/HTTPS che non rientrano nelle casistiche consentite. 
Ad esempio droppare tutte le richieste web che non contengono nell'header "Host" il dominio o i domini presenti nei certificati Let'sEncrypt, e quindi droppare le varie botnet o crawler che scansionano la Rete alla ricerca di servizi noti da poter violare. 
Inoltre puoi loggare le richieste provenienti da rete pubblica (sempre tramite reverse proxy) e attivare una notifica su HA che ti avvisi quando qualcuno o qualcosa ha provato a contattare il tuo router.
Però preoccuparsi di rimuovere il port-mapping HTTP(80) quando stai comunque esponendo su internet tutta la tua domotica (spero dietro 2FA), mi sembra una precauzione superflua. Io sono dell'idea che i sistemi vadano resi sicuri, non inaccessibili. 
Sempre consapevoli che non esistono sistemi sicuri al 100% e che quello che possiamo fare è rendere la vita complicata ai malintenzionati.

Quanto al problema del listener già attivo sulla porta 80, non conoscendo il tuo sistema e la tua configurazione posso solo fare congetture e ipotizzare che sia rimasto appeso qualche processo del proxy dall'ultimo rinnovo del certificato, ma immagino avrai già provato a riavviare il tuo reverse proxy, e in ultima analisi il sistema, minipc, raspberry, o NAS che sia.
Però mi aspetterei che sullo stesso reverse proxy fosse attivo, sia il listener per HA, sia quello per Let'sEncrypt, quindi, una volta riattivato il port-forwarding sul router, non dovresti dover fare nient'altro e HTTP(80) dovrebbe essere subito raggiungibile. Sono un po' perplesso, non capisco com'è configurato il tuo ambiente
